src/backend/main.ts

```diff
@@ -200,132 +205,151 @@ export async function updateOpenDataIndex() {
   const domain = process.env.DOMAIN_NAME_OPEN_DATA;
   if (!domain) throw new Error('Open data domain name missing from environment');
 
-  //
-  // Fetch data files
-
-  // TODO: Model these (meta: OpenDataMeta, data: T = any)
-  const openDataIndex: Record<string, any> = {};
-
-  for (const filename of openDataFileNames) {
-    const data = await s3GetJsonHelper({ Bucket: bucket, Key: `${filename}.json` });
-    openDataIndex[filename] = data.meta;
-  }
-
-  const openData = {
-    meta: {
-      description: 'This is the open data site for the www.oiretutka.fi project',
-    },
-    data: openDataIndex,
-  };
-
-  await s3PutJsonHelper({ Bucket: bucket, Key: 'index.json', Body: openData });
-}
-
-async function mapPostalCodeLevelToCityLevelData(postalCodeLevelData: any[], bucket: string) {
-  const postalCodeCityMappings = await s3GetJsonHelper({ Bucket: bucket, Key: 'postalcode_city_mappings.json' });
-
-  const populationPerCity = await s3GetJsonHelper({ Bucket: bucket, Key: 'population_per_city.json' });
-
-  // Generate initial city-level aggregation set from the city population data
-  const resultsByCity = (populationPerCity.data as any[]).reduce((acc, data) => {
-    acc[data.city] = {
-      city: data.city,
-      population: data.population,
-      responses: 0,
-      fever_no: 0,
-      fever_slight: 0,
-      fever_high: 0,
-      cough_no: 0,
-      cough_mild: 0,
-      cough_intense: 0,
-      general_wellbeing_fine: 0,
-      general_wellbeing_impaired: 0,
-      general_wellbeing_bad: 0,
-      breathing_difficulties_no: 0,
-      breathing_difficulties_yes: 0,
-      muscle_pain_no: 0,
-      muscle_pain_yes: 0,
-      headache_no: 0,
-      headache_yes: 0,
-      sore_throat_no: 0,
-      sore_throat_yes: 0,
-      rhinitis_no: 0,
-      rhinitis_yes: 0,
-      stomach_issues_no: 0,
-      stomach_issues_yes: 0,
-      sensory_issues_no: 0,
-      sensory_issues_yes: 0,
-      longterm_medication_no: 0,
-      longterm_medication_yes: 0,
-      smoking_no: 0,
-      smoking_yes: 0,
-      corona_suspicion_no: 0,
-      corona_suspicion_yes: 0,
+  return Promise.all(
+    //
+    // Fetch data files
+    openDataFileNames.map(filename =>
```

Comment:
Big difference here to `async` version is that the s3 resources are loaded in parallel here. Performance gain is probably neglible, does take a small memory hit though.